server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexTemplateService.java

```diff
@@ -281,6 +304,48 @@ static ClusterState addIndexTemplateV2(final ClusterState currentState, final bo
             .build();
     }
 
+    /**
+     * Return a map of v1 template names to their index patterns for v1 templates that would overlap
+     * with the given v2 template's index patterns.
+     */
+    static Map<String, List<String>> findConflictingV1Templates(final ClusterState state, final String candidateName,
+                                                                final List<String> indexPatterns) {
+        Automaton v2automaton = Regex.simpleMatchToAutomaton(indexPatterns.toArray(Strings.EMPTY_ARRAY));
```

Comment:
👍 This method makes validating overlapping templates really easy for index templates v2 :)